### PR TITLE
Add .netrc binding

### PR DESCRIPTION
**Pull request by silvio**
Signed-off-by: Silvio Fricke <silvio.fricke@gmail.com>

### CODE REVIEW

**Comment by silvio** [comment]
@alexcrichton I'm not sure why my changes crashes at systest time. I'm not know this. What I have to do to get it compiled/tested?

**Comment by alexcrichton** [comment]
@silvio oh that typically means that the sys bindings differ from curl itself. You should be able to reproduce locally with a `cargo run` in the systest dir

**Comment by silvio** [comment]
@alexcrichton okay thanks. I have fixed this problem.

File: src/easy.rs [hunk 1]
```diff
@@ -307,6 +307,24 @@ pub enum WriteError {
     __Nonexhaustive,
 }
 
+/// Options for `.netrc` parsing.
+pub enum CurlNetrcOption {
```
Comment (alexcrichton) [inline src/easy.rs]: Perhaps this could be called just 'NetRc'?

File: src/easy.rs [hunk 2]
```diff
@@ -307,6 +307,24 @@ pub enum WriteError {
...
+    /// Ignoring `.netrc` file and use information from url
+    ///
+    /// This option is default
+    Ignored,
```
Comment (alexcrichton) [inline src/easy.rs]: Could you set the discriminants here to the sys values to ensure they match up?

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

**Comment by alexcrichton** [comment]
er

**Comment by silvio** [comment]
thanks!

**Comment by silvio** [comment]
@alexcrichton  I'm depending on this feature. What is your time planing for publishing a curl-rust version with this feature?

**Comment by alexcrichton** [comment]
Oh I tend to just release whenever there's a request, so I'll make a release now!

**Pull request merged**